kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/EventHandlerServlet.java

```diff
@@ -81,14 +88,20 @@ public void handleEvent(final Event event) {
     public void doGet(final HttpServletRequest request, final HttpServletResponse response)
             throws ServletException, IOException {
 
+        // if the queue is full, clear it as we don't need to care about rejected events
+        if (this.sharedCache.size() == MAX_SIZE_OF_QUEUE) {
+            this.sharedCache.clear();
```

Comment:
If you clear the queue here. It will be empty a few lines below. Why that? Is the upper bound not enough?